scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5258 @@
+
+    def draw(self, sizes=None, rng=None, proportions=None, region='domain'):
+        r"""Draw random values of all parameters for use in testing.
+
+        Parameters
+        ----------
+        sizes : iterable of shape tuples
+            The size of the array to be generated for each parameter in the
+            parameterization. Note that the order of sizes is arbitary; the
+            size of the array generated for a specific parameter is not
+            controlled individually as written.
+        rng : NumPy Generator
+            The generator used to draw random values.
+        proportions : tuple
+            A tuple of four non-negative numbers that indicate the expected
+            relative proportion of elements that are within the parameter's
+            domain, are on the boundary of the parameter's domain, are outside
+            the parameter's domain, and have value NaN. For more information,
+            see the `draw` method of the _Parameter subclasses.
+        domain : str
+            The domain of the `_Parameter` from which to draw. Default is
+            "domain" (the *full* domain); alternative is "typical".
+
+        Returns
+        -------
+        parameter_values : dict (string: array)
+            A dictionary of parameter name/value pairs.
+        """
+        # ENH: be smart about the order. The domains of some parameters
+        # depend on others. If the relationshp is simple (e.g. a < b < c),
+        # we can draw values in order a, b, c.
+        parameter_values = {}
+
+        if not len(sizes) or not np.iterable(sizes[0]):
+            sizes = [sizes]*len(self.parameters)
+
+        for size, param in zip(sizes, self.parameters.values()):
+            parameter_values[param.name] = param.draw(
+                size, rng=rng, proportions=proportions,
+                parameter_values=parameter_values,
+                region=region
+            )
+
+        return parameter_values
+
+
+def _set_invalid_nan(f):
+    # Wrapper for input / output validation and standardization of distribution
+    # functions that accept either the quantile or percentile as an argument:
+    # logpdf, pdf
+    # logcdf, cdf
+    # logccdf, ccdf
+    # ilogcdf, icdf
+    # ilogccdf, iccdf
+    # Arguments that are outside the required range are replaced by NaN before
+    # passing them into the underlying function. The corresponding outputs
+    # are replaced by the appropriate value before being returned to the user.
+    # For example, when the argument of `cdf` exceeds the right end of the
+    # distribution's support, the wrapper replaces the argument with NaN,
+    # ignores the output of the underlying function, and returns 1.0. It also
+    # ensures that output is of the appropriate shape and dtype.
+
+    endpoints = {'icdf': (0, 1), 'iccdf': (0, 1),
+                 'ilogcdf': (-np.inf, 0), 'ilogccdf': (-np.inf, 0)}
+    replacements = {'logpdf': (-inf, -inf), 'pdf': (0, 0),
```

Comment:
```suggestion
    replacements = {'logpdf': (-inf, 0), 'pdf': (0, 1),
```

Reply:
I think this is right as is. What it's saying is that to the left or right of the support, the pdf is 0.

Reply:
I mixed up pdf snd cdf.